### ISSUE TITLE
Annotated[..., Type] forward references are not understood

### ISSUE THREAD

**Issue by treykeown** [microsoft/pylance-release#4565]
## Environment data

-   Language Server version: 2023.6.40
-   OS and version: macOS 13.4.1
-   Python version: 3.11.4

## Code Snippet

```python
from __future__ import annotations
from typing import Annotated, get_type_hints

class Foo:
    f: Annotated[str, Bar] = "asdf"

class Bar: ...

print(get_type_hints(Foo, include_extras=True))
```

```console
$ python3 example.py 
{'f': typing.Annotated[str, <class '__main__.Bar'>]}
$ mypy example.py 
Success: no issues found in 1 source file
```

<img width="618" alt="issue2" src="https://github.com/microsoft/pylance-release/assets/2755914/bf9b39c0-e11e-4665-befe-9e311cb1efb6">

## Expected behavior

`Bar` is understood to be a forward reference

## Actual behavior

`Bar` is evaluated like a normal value

## Logs

N/A?

**Comment by StellaHuang95** [comment]
From what I understand, if you use `from __future__ import annotations`, you don't need to quote forward references, it will postpone the evaluation of the annotation. But since in `Annotated[T, x]`, `x` is metadata not a type annotation, not sure if  `__future__` should cause that to be ignored or not. Tagging @erictraut for further insights.

**Comment by erictraut** [comment]
Currently, pyright allows forward references only with type annotations. It doesn't allow this for non-type annotations (i.e. those that appear after the type annotation when using `Annotated`). To be consistent with the runtime, it would be more correct to apply deferred annotation evaluation for non-type annotations as well when `from __future__ import annotations` is used. I've made this change in the pyright source base.

**Comment by heejaechang** [comment]
This issue has been fixed in prerelease version 2023.7.31, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#2023731-19-July-2023-PreRelease)
